VL.ImGui.Stride/src/ImGuiRenderer.cs

```diff
@@ -0,0 +1,205 @@
+﻿using ImGuiNET;
+using Stride.Input;
+using Stride.Graphics;
+using Stride.Rendering;
+using Stride.Core.Mathematics;
+using System.Runtime.CompilerServices;
+using Buffer = Stride.Graphics.Buffer;
+using RendererBase = Stride.Rendering.RendererBase;
+
+using VL.Core;
+using VL.Lib.Collections;
+using VL.Stride;
+using VL.Stride.Rendering;
+using VL.Stride.Input;
+
+using System.Reactive.Disposables;
+
+namespace VL.ImGui
+{
+    using ImGui = ImGuiNET.ImGui;
+
+    public partial class ImGuiRenderer : RendererBase, IDisposable
+    {
+        const int INITIAL_VERTEX_BUFFER_SIZE = 128;
+        const int INITIAL_INDEX_BUFFER_SIZE = 128;
+
+        // dependencies
+        GraphicsDevice device;
+        GraphicsContext context;
+        CommandList commandList;
+        InputManager input;
+
+        // ImGui
+        private readonly ImGuiIOPtr _io;
+        private readonly Context _context;
+        private ImDrawDataPtr _drawDataPtr;
+        private float _fontScaling;
+        private float _uiScaling;
+        private bool _readyToBeDrawn;
+
+        private Widget widget;
+        private WidgetLabel widgetLabel = new();
+        private bool dockingEnabled;
+        private Spread<FontConfig> fonts = Spread.Create(FontConfig.Default);
+        private bool fullscreenWindow;
+        private IStyle style;
+
+        // Stride
+        private PipelineState imPipeline;
+        private VertexDeclaration imVertLayout;
+        private VertexBufferBinding vertexBinding;
+        private IndexBufferBinding indexBinding;
+        private CustomDrawEffect imShader;
+        private Texture fontTexture;
+        private Matrix projMatrix;
+
+        private IInputSource lastInputSource;
+        private readonly SerialDisposable inputSubscription = new SerialDisposable();
+        public unsafe ImGuiRenderer(CustomDrawEffect drawEffect)
+        {
+            device = AppHost.Current.Services.GetDeviceHandle().Resource;
```

Comment:
You always need to store the `IResourceHandle<T>` and dispose it in `Dispose`. Otherwise on shutdown we have a leak here because the handle is not returned and therefor depending resources won't be able to shutdown.